File: solutions/python/grains/1/grains.py
def square(number):
    if number < 1 or number > 64:
        raise ValueError("square must be between 1 and 64")
        
    return 2 ** (number - 1)

def total():
    grain_num = 0

    for i in range(1, 65):
        grain_num += square(i)

    return grain_num

File: solutions/kotlin/grains/1/src/main/kotlin/Grains.kt
import java.math.BigInteger
import kotlin.math.pow


object Board {

    fun getGrainCountForSquare(number: Int): BigInteger {
        if(number !in 1..64) throw IllegalArgumentException("Number must be in range 1..64")

        return BigInteger.TWO.pow(number - 1)
    }

    fun getTotalGrainCount(): BigInteger {
        return BigInteger.TWO.pow(64) - BigInteger.ONE
    }
}
